### DISCUSSION TITLE
Problem with resnet50 build for Alveo

### DISCUSSION THREAD

**Discussion by Galapple** [Xilinx/finn#1013] [Q&A] [answered]
Hi, I am trying to build the design for the Alveo U250 with 3_1 shell.
I have set the enviroment as decribed (I think) but get this error when runing the command:
`./run-docker.sh build_custom $FINN_EXAMPLES/build/resnet50`

**These are the errors:**
Traceback (most recent call last):
  File "/home/gal/finn_1/finn-examples/build/finn/src/finn/builder/build_dataflow.py", line 168, in build_dataflow_cfg
    model = transform_step(model, cfg)
  File "/home/gal/finn_1/finn-examples/build/resnet50/custom_steps.py", line 224, in step_resnet50_convert_to_hls
    model = model.transform(trn())
  File "/home/gal/finn_1/finn-examples/build/finn/deps/qonnx/src/qonnx/core/modelwrapper.py", line 140, in transform
    (transformed_model, model_was_changed) = transformation.apply(transformed_model)
  File "/home/gal/finn_1/finn-examples/build/finn/src/finn/transformation/fpgadataflow/convert_to_hls_layers.py", line 1243, in apply
    model = model.transform(InferShapes())
  File "/home/gal/finn_1/finn-examples/build/finn/deps/qonnx/src/qonnx/core/modelwrapper.py", line 140, in transform
    (transformed_model, model_was_changed) = transformation.apply(transformed_model)
  File "/home/gal/finn_1/finn-examples/build/finn/deps/qonnx/src/qonnx/transformation/infer_shapes.py", line 88, in apply
    hidden_ops = _hide_finn_ops(model)
  File "/home/gal/finn_1/finn-examples/build/finn/deps/qonnx/src/qonnx/transformation/infer_shapes.py", line 60, in _hide_finn_ops
    new_node = _make_shape_compatible_op(node, model)
  File "/home/gal/finn_1/finn-examples/build/finn/deps/qonnx/src/qonnx/transformation/infer_shapes.py", line 44, in _make_shape_compatible_op
    inst = registry.getCustomOp(node)
  File "/home/gal/finn_1/finn-examples/build/finn/deps/qonnx/src/qonnx/custom_op/registry.py", line 45, in getCustomOp
    inst = inst_wrapper(node, onnx_opset_version=onnx_opset_version)
TypeError: __init__() got an unexpected keyword argument 'onnx_opset_version'
> /home/gal/finn_1/finn-examples/build/finn/deps/qonnx/src/qonnx/custom_op/registry.py(45)getCustomOp()
-> inst = inst_wrapper(node, onnx_opset_version=onnx_opset_version)
(Pdb)

**Comment by mariodruiz** [comment]
Can you please described the steps you took?

**Comment by Galapple** [comment]
Hi Mario,

1. I have created a docker container
2. Clone the FINN and the FINN examples repositories
3. installed conda
4. Changed all env variables as described here
https://finn.readthedocs.io/en/latest/getting_started.html#requirements
5. Run this command to build the resnet50 file :./run-docker.sh
build_custom $FINN_EXAMPLES/build/resnet50

The build starts and stops at step 7 (step_hls_codegen [7/13]) with these
warnings:
WARNING: Unable to find /settings64.sh
WARNING: Functionality dependent on Vitis will not be available.
WARNING: If you need Vitis, ensure VITIS_PATH is set correctly and mounted
into the Docker container.
WARNING: Unable to find /settings64.sh
WARNING: Functionality dependent on Vivado will not be available.
WARNING: If you need Vivado, ensure VIVADO_PATH is set correctly and
mounted into the Docker container.
WARNING: Unable to find /settings64.sh
WARNING: Functionality dependent on Vitis HLS will not be available.
WARNING: Please note that FINN needs at least version 2020.2 for Vitis HLS
support.
WARNING: If you need Vitis HLS, ensure HLS_PATH is set correctly and
mounted into the Docker container.

**Reply by mariodruiz** [reply]
which version of Vitis do you have installed?

**Comment by Galapple** [comment]
2021.1

**Comment by mariodruiz** [comment]
Hi @Galapple,

By any chance are you running the docker container inside a conda env.
For your information, the development phase should only run on docker, no conda.

For, deployment you will need the docker container.

More advice,

I would recommend that you use finn-examples **dev** branch and use the `get-finn.sh` script like described in the README of finn-examples: https://github.com/Xilinx/finn-examples/tree/dev/build
Also, I would suggest you use Vivado 2022.2

**Comment by Galapple** [comment]
I have tried it with deactivating the conda env and I have got this errors:
Running step: step_hls_ipgen [8/13]
multiprocessing.pool.RemoteTraceback:
"""
Traceback (most recent call last):
  File "/opt/conda/lib/python3.8/multiprocessing/pool.py", line 125, in
worker
    result = (True, func(*args, **kwds))
  File "/opt/conda/lib/python3.8/multiprocessing/pool.py", line 48, in
mapstar
    return list(map(*args))
  File
"/home/gal/finn_1/finn-examples/build/finn/src/finn/transformation/fpgadataflow/hlssynth_ip.py",
line 73, in applyNodeLocal
    inst.ipgen_singlenode_code()
  File
"/home/gal/finn_1/finn-examples/build/finn/src/finn/custom_op/fpgadataflow/hlscustomop.py",
line 358, in ipgen_singlenode_code
    assert os.path.isdir(
AssertionError: IPGen failed:
/tmp/finn_dev_gal/code_gen_ipgen_ChannelwiseOp_Batch_2_8iis4ywo/project_ChannelwiseOp_Batch_2/sol1/impl/ip
not found. Check log under
/tmp/finn_dev_gal/code_gen_ipgen_ChannelwiseOp_Batch_2_8iis4ywo
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File
"/home/gal/finn_1/finn-examples/build/finn/src/finn/builder/build_dataflow.py",
line 168, in build_dataflow_cfg
    model = transform_step(model, cfg)
  File
"/home/gal/finn_1/finn-examples/build/finn/src/finn/builder/build_dataflow_steps.py",
line 493, in step_hls_ipgen
    model = model.transform(HLSSynthIP())
  File
"/home/gal/finn_1/finn-examples/build/finn/deps/qonnx/src/qonnx/core/modelwrapper.py",
line 140, in transform
    (transformed_model, model_was_changed) =
transformation.apply(transformed_model)
  File
"/home/gal/finn_1/finn-examples/build/finn/deps/qonnx/src/qonnx/transformation/base.py",
line 111, in apply
    new_nodes_and_bool = p.map(self.applyNodeLocal, old_nodes, chunksize=1)
  File "/opt/conda/lib/python3.8/multiprocessing/pool.py", line 364, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/opt/conda/lib/python3.8/multiprocessing/pool.py", line 771, in get
    raise self._value
AssertionError: IPGen failed:
/tmp/finn_dev_gal/code_gen_ipgen_ChannelwiseOp_Batch_2_8iis4ywo/project_ChannelwiseOp_Batch_2/sol1/impl/ip
not found. Check log under
/tmp/finn_dev_gal/code_gen_ipgen_ChannelwiseOp_Batch_2_8iis4ywo
> /opt/conda/lib/python3.8/multiprocessing/pool.py(771)get()
-> raise self._value

Should I upgrade the vitis to 2022.2 or just the vivado?

**Comment by mariodruiz** [comment, marked as answer]
Hi @Galapple,

Please start in a new terminal and using the finn-examples dev branch.

>Should I upgrade the vitis to 2022.2 or just the vivado?

Vitis
